### PR TITLE
#164 allow tagging of discord users in username field

### LINKED ISSUE #164: [Feature Suggestion] Allow tagging discord users in "username" fields
It would be nice if `/stats username:@Psikoi` (discord user tag) worked, for scenarios when you don't know your friend's current username.

It would try and fetch the current alias (from our database) to that discord tag, and use it as the username parameter for the command. Not too different from the way we currently "infer" usernames when they are not provided.

### Currently

`getUsernameParam`
- if has username param, use that
- otherwise, use the alias for the user that called the command

### Ideal

`getUsernameParam`
- if has username option
     - if it's a user tag `<@userId>`, use the alias for that userId
     - otherwise, use that username option value as the username for the WOM request
- otherwise, use the alias for the user that called the command

### PR DESCRIPTION
**Pull request by aleccaputo**
If a string username is passed, use that
If a discord tag is passed (@ Foo) try and fetch the alias for that discord id and user that If no param is passed, infer the alias for the user that sent the message

Fixes #164

### CODE REVIEW

**Comment by aleccaputo** [comment]
Not sure if any of these regexes existed elsewhere in the solution, feel free to point me there if so

File: src/utils/commands.ts [hunk 1]
```diff
@@ -165,10 +165,16 @@ function attachOptions(
 }
 
 export async function getUsernameParam(interaction: CommandInteraction) {
+  const discordTagRegex = /<@!?(\d+)>/;
```
Comment (psikoi) [inline src/utils/commands.ts:168]: I think it would be best to move this as a constant to the top of the file `DISCORD_TAG_REGEX`

**Review by psikoi — CHANGES_REQUESTED** [review]

File: src/utils/commands.ts [hunk 2]
```diff
@@ -165,10 +165,16 @@ function attachOptions(
...
   const username = interaction.options.getString('username', false);
-  if (username) return username;
+  const isDiscordId = username.match(discordTagRegex);
```
[the remark this answers is not in the document]
Comment (aleccaputo) [reply src/utils/commands.ts:170]: yeah good call, weird that ts inferred that to be no a nullable string when the return type is clearly `string | null` from discordjs
image

Comment (Jonxslays) [inline src/utils/commands.ts:170]: This doesnt error when no username is passed as an option?

[the remark this answers is not in the document]
Comment (aleccaputo) [reply src/utils/commands.ts]: image

Comment (aleccaputo) [reply src/utils/commands.ts]: fixed. weird TS inference

File: src/utils/commands.ts (hunk 1 above)
Comment (aleccaputo) [reply src/utils/commands.ts]: done

File: src/utils/commands.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (aleccaputo) [reply src/utils/commands.ts]: updated

**Comment by aleccaputo** [comment]
ready for re-review

**Pull request merged**